lang/funcs/operator_polyfunc.go

```diff
@@ -169,9 +172,14 @@ func init() {
 	RegisterOperator("==", &types.FuncValue{
 		T: types.NewType("func(a float, b float) bool"),
 		V: func(input []types.Value) (types.Value, error) {
-			// TODO: should we do an epsilon check?
+			// epsilon check
+			if math.Abs(input[0].Float()-input[1].Float()) <= epsilon {
```

Comment:
I think the more elegant way to write this is to start off with something like:

`result := false`

And then in you condition you do: `result = true`

And at the end you put in the bool into the &BoolValue as result. (Similarly for other cases.)